src/main/java/com/example/eventdrivennasa/service/PictureService.java

```diff
@@ -0,0 +1,64 @@
+package com.example.eventdrivennasa.service;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.example.eventdrivennasa.controller.NasaFeignClient;
+import com.example.eventdrivennasa.model.LargestPicRequestDTO;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class PictureService {
+
+    Map<String, byte[]> picStorage = new ConcurrentHashMap<>();
```

Comment:
Since you have some checks/validations in the `findPicture()` method, it would be good to make this map **private**, otherwise anyone can add or get items from your picStorage ignoring your checks. And it's generally a good practice to make fields responsible for the internal logic **private**.